src/adapter/src/coord/sequencer.rs

```diff
@@ -2232,169 +2232,166 @@ impl<S: Append + 'static> Coordinator<S> {
         session: &Session,
         plan: ExplainPlanNew,
     ) -> Result<ExecuteResponse, AdapterError> {
+        use mz_compute_client::plan::Plan;
+        use mz_ore::tracing::trace_plan;
+        use ExplainStageNew::*;
+
         let compute_instance = self
             .catalog
             .resolve_compute_instance(session.vars().cluster())?
             .id;
 
         let ExplainPlanNew {
-            mut raw_plan,
+            raw_plan,
             row_set_finishing,
             stage,
             format,
             config,
             explainee,
         } = plan;
 
-        let decorrelate = |raw_plan: HirRelationExpr| -> Result<MirRelationExpr, AdapterError> {
-            let decorrelated_plan = raw_plan.optimize_and_lower(&OptimizerConfig {
-                qgm_optimizations: session.vars().qgm_optimizations(),
-            })?;
-            Ok(decorrelated_plan)
-        };
+        let optimizer_trace = OptimizerTrace::new();
+
+        let (used_indexes, fast_path_plan) =
+            optimizer_trace.collect_trace(|| -> Result<_, AdapterError> {
+                let raw_plan = raw_plan.clone(); // FIXME: remove `.clone()` once the QGM Model implements Clone.
+                let _span = tracing::span!(Level::INFO, "optimize").entered();
+
+                tracing::span!(Level::INFO, "raw").in_scope(|| {
+                    trace_plan(&raw_plan);
+                });
```

Comment:
i like this, the `raw` and `optimize` next to each other!

I'm not an expert so i don't know what `local` means, but the tracing components of this change make sense!